// mymodule/6.js
import "https://cdn.jsdelivr.net/npm/dayjs@1/dayjs.min.js"
import "https://cdn.jsdelivr.net/lodash/4.10.0/lodash.js"



let resarray =[];
let result =[];
let arrayfromrand=[];

function course()
{
   const ar=["Mathematics", "Physics", "English", "Computer Science", "Dancing", "Chess", "Biology", "Chemistry", 
      "Law", "Art", "Medicine", "Statistics"]; 
   //a=0 b=11
   let im=Math.floor(Math.random()*(11-0+1))+0;
   return ar[im];
}

function addone(res){
   arrayfromrand.push(res);
}

async function task1(kol)
{
 
   for(let i = 0; i < kol ; i ++)
   {
      let results = await fetch('https://randomuser.me/api')
         .then((response) => response.json())
         .then((json) => addone(json));
      
   }


   const array1 = arrayfromrand;//Array.from(name.randomUserMock);
   const array3 =[];
for(let i = 0; i < array1.length ; i ++)
{
   let a= array1[i].results[0];
   let b={};
   b.gender=a.gender;
   b["title"]=a.name.title;
   b.full_name=a.name.first + " " + a.name.last;
   b.city=a.location.city;
   b.state=a.location.state;
   b.country=a.location.country;
   b.postcode=a.location.postcode;
   b.coordinates=a.location.coordinates;
   b.timezone=a.location.timezone;
   b.email=a.email;
   b.b_date=a.dob.date;
   b.age=a.dob.age;
   b.phone=a.phone;
   b.picture_large=a.picture.large;
   b.picture_thumbnail=a.picture.thumbnail;   
   b.id=a.id.name+a.id.value;
   array3[i]=b;
}

for(let i = 0; i < array3.length ; i ++)
{
   
   array3[i].favorite=false;
   array3[i].course=null;//course();
   array3[i].bg_color="#dface7";
   array3[i].note="Note";
}


let array_result=array3;

for(let i = 0; i < array_result.length ; i ++)
{
   for(let j = i+1; j < array_result.length ; j ++)
   {
    
      if(array_result[i].full_name==array_result[j].full_name){array_result.splice (j, 1);}
      
   }
  if (array_result[i].course==null){array_result[i].course=course();}
  resarray.push(array_result[i]);
}

}   

function check(a,por)
{
   if (por==1){
      if(a==undefined) return false;
      if(a==null) return false;
      if(typeof(a)!="string") return false;
      if(a.substring(0,1)==a.substring(0,1).toUpperCase()) return true;
   }
   if (por==2){
      if(a==undefined) return false;
      if(a==null) return false;
      if(typeof(a)!="string") return false;
      return true;
   }
   if (por==3){
      if(a==undefined) return false;
      if(a==null) return false;
      if(typeof(a)!="number") return false;
      return true;
   }
   if (por==4){
      if(a==undefined) return false;
      if(a==null) return false;
      if(typeof(a)!="string") return false;
      const exp = /@/;
      return exp.test (a);
   }
}



function task2(){


   let ar = resarray;
   for(let i = 0; i < ar.length ; i ++)
   {
    
       let a=ar[i].full_name;
       let b=ar[i].state;
       let c=ar[i].city
       let d=ar[i].country;
       let e=ar[i].note;
       let f=ar[i].gender;
       let g=ar[i].age;
       let m=ar[i].email;
       let result=false;
       if ( check(a,1)&&check(b,1)&&check(c,1)&&check(d,1)&&check(e,1)&&check(f,2)&&check(g,3)&&check(m,4))
       {result=true;}
       ar[i].check=result;
       
   }
}

export function task4(field)
{
   resarray=_.sortBy(resarray, function (item) {
      if (field==0) {return item.full_name.toLowerCase();}
      else if (field==1) {return item.course.toLowerCase();}
      else if (field==2) {return item.age;}
      else if (field==3) {return item.gender.toLowerCase();}
      else if (field==4) {return item.country.toLowerCase();}
    })
}



export function task5(value)
{  
   task8();
   const ar=result;
   console.log(value)
   const ob1=_.find(ar,function(ob){return ob.note==value||ob.age==value||ob.full_name==value})
   result=[];
   result.push(ob1);
  
  
   task7();
   
}



async function task7()
{
   let ar=result;
   var box = document.getElementById("gallery"); 
   let txt='';

   for(let i = 0; i < ar.length ; i ++)
   {
        let res=ar[i];
            txt+='<div class="teacher">';
            txt+='<a href="#compact-card-popup'+i+'">';

            txt+=' <div class="img"><img src="'+res.picture_large+'" alt=" Фото викладача "  width=200 height=200>';
            txt+='</div>';
            txt+='</a>';
            txt+='<h2>'+res.full_name+'</h2>';
            txt+='<label>'+res.course+'</label>';
            txt+='<label>'+res.country+'</label>';
            txt+='</div>';

    txt+='<div id="compact-card-popup'+i+'" class="modal">';
    txt+='    <div class="content">';
    txt+='        <header> ';
    txt+='           <h2>Compact card</h2>';
    txt+='            <a href="#" class="box-close">x</a>';
    txt+='        </header>';
    txt+='        <form>';
    txt+='            <figure>';
    txt+='                <img src="'+res.picture_large+'"/>'
    txt+='                <figcaption>'
    txt+='                    <output name="full-name-out">'+res.full_name+'</output>'
    txt+='                </figcaption>'
    txt+='            </figure>'
    txt+='            <a class="button" href="#full-card-popup'+i+'">';
    txt+='                Open full card';
    txt+='            </a>';
    txt+='        </form>';
    txt+='    </div>';
    txt+='</div>';




       txt+='<div id="full-card-popup'+i+'" class="modal">';

       txt+=' <div class="content">'
       txt+='     <header>'
       txt+='         <h2>Full card</h2>'
       txt+='         <a href="#" class="box-close">x</a>'
       txt+='     </header>'

       txt+='     <form>'
       txt+='         <section>'
       txt+='             <label>Full name</label>'
       txt+='             <output>'+res.full_name+'</output>'
       txt+='         </section>'
                
       txt+='         <section>'
       txt+='             <label>Sex</label>'
       txt+='             <output>'+res.gender+'</output>'
       txt+='         </section>'
       
       
       txt+='         <div id="datecontact">'
       txt+='         <section>'
       txt+='             <label>Date of birth</label>'
       txt+='             <output>'+res.b_date+'</output>'
       txt+='         </section>'

       txt+='         <section>'
       txt+='             <label>Day to Birthday</label>'
       txt+='             <output>'+dayforbirthday(res.b_date)+'</output>'
       txt+='         </section>'
       txt+='         </div>'
                
       txt+='         <div id="contact">'
       txt+='             <section>'
       txt+='                 <label>e-mail</label>'
       txt+='                 <output>'+res.email+'</output>'
       txt+='             </section>'
        
       txt+='             <section>'
       txt+='                 <label>Telephone</label>'
       txt+='                 <output>'+res.phone+'</output>'
       txt+='             </section>'
       txt+='         </div>'
                
       txt+='         <div id="location">'
        txt+='            <section>'
       txt+='                 <label>Country</label>'
       txt+='                 <output>'+res.country+'</output>'
       txt+='             </section>'
        
       txt+='             <section>'
       txt+='                <label>City</label>'
       txt+='                 <output>'+res.city+'</output>'
       txt+='             </section>'
       txt+='        </div>'
    
       txt+='         <section>'
       txt+='             <label>Comment</label>'
       txt+='             <output>'+res.note+'</output>'
       txt+='         </section>'

         txt+='         <section>'

    txt+='<button onclick="window.open(\'table6.html?par1='+res.coordinates.latitude+'&par2='+res.coordinates.longitude+'\', \'_blank\',\'top=100, left=100, width=600, height=400\');">Open Map!</button>' 
     txt+='         </section>'

       txt+='     </form>'

       txt+=' </div>'

       txt+='</div>'
 }
 box.innerHTML = txt;

 var box1 = document.getElementById("grid"); 

 let txt1='<tr>'
 txt1+='<th>Name</th>'
 txt1+='<th>Speciality</th>'
 txt1+='<th>Age</th>'
 txt1+='<th>Sex</th>'
 txt1+='<th>Nationality</th>'
 txt1+='</tr>'

 txt1+='<tbody>'
 for(let i = 0; i < ar.length ; i ++)
 {
      let res=ar[i];
      txt1+='<tr>'
      txt1+='<td>'+res.full_name+'</td>'
      txt1+='<td>'+res.course+'</td>'
      txt1+='<td>'+res.age+'</td>'
      txt1+='<td>'+res.gender+'</td>'
      txt1+='<td>'+res.country+'</td>'
      txt1+='</tr>'
 }
 txt1+='</tbody>'

 txt1+='<div>'
  txt1+='   <h1>Кількість по країнам</h1>'
  txt1+='  <canvas id="grafica"></canvas>'

  txt1+='<script  type = "module" >' 
  txt1+='import "https://cdn.jsdelivr.net/npm/chart.js@latest/dist/Chart.min.js"'
   txt1+="import {piechat} from  './6.js';"
 txt1+='piechat();'
 txt1+='</script>'
 txt1+='</div>'
 box1.innerHTML = txt1;



 var box2 = document.getElementById("slider"); 
 let txt2=            '<div class="container">'
 txt2+='<a id="prev" class="slide-btn">&#10094;</a>'
 txt2+='<div class="items-container">'
 for(let i = 0; i < ar.length ; i ++)
 {
    let res=ar[i];
    txt2+='<div class="teacher" class="entry">'
    txt2+='<a href="#compact-card-popup'+i+'">';
    //txt2+='<a href="#compact-card-popup">'
    txt2+='<div class="img"><img src="'+res.picture_thumbnail+'" alt=" Фото викладача "  width=200 height=200>'
    txt2+='</div>'
    txt2+='</a>'
    txt2+='<h2>'+res.full_name+'</h2>'
    txt2+='<label>'+res.country+'</label>'
    txt2+='</div>'
 }
 txt2+='</div>'
 txt2+='<a id="next" class="slide-btn">&#10095;</a>'
 txt2+='</div>'


 box2.innerHTML = txt2;

}



function task8(){ 
 
   result=resarray.slice ();;//.slice();
   var x = document.getElementById("age").selectedIndex;
   let res=result;
   if (x=="1"){const res1 = _.filter (res,function(country){return country.age>=18&&country.age<=31}); result=res1;} 
   if (x=="2"){const res1 = res.filter (country => country.age>=32&&country.age<=58); result=res1;} 
   if (x=="3"){const res1 = res.filter (country => country.age>=59&&country.age<=75); result=res1;} 
   var z = document.getElementById("region").selectedIndex;
   res=result;
   if (z=="1"){const res1 = res.filter (country => country.country=="Україна"); result=res1;} 
   if (z=="2"){const res1 = res.filter (country => country.country=="Germany"); result=res1;} 
   if (z=="3"){const res1 = res.filter (country => country.country=="Norway"); result=res1;} 
   res=result;
   var y = document.getElementById("sex").selectedIndex;
   if (y=="1"){const res1 = res.filter (country => country.gender=="male"); result=res1;} 
   if (y=="2"){const res1 = res.filter (country => country.gender=="female"); result=res1;} 
  
   if (document.getElementById("photo").checked==true){
      res=result;
      const res1 = res.filter (country =>/*country.picture_thumbnail.length!=0 */country.picture_thumbnail!= undefined && country.picture_thumbnail!=null); 
      result=res1;
   }

   if (document.getElementById("favourite").checked==true){
      res=result;
      const res1 = res.filter (country =>country.favorite!= undefined && country.favorite!=null && country.favorite==true); 
      result=res1;
   }
 
};


export async function start(){
  
await task1(50);
task2();
task8();
await task7();
}


export function next(){
 
   task8();
   task7();
   }

export function add(name,specialization,email,tel,region,city,sex,date,color,comment)
{
   let b={};
   
   b.gender=sex;
   b.full_name=name
   b.city=city;
   if (region="ua") {b.country="Україна";}
   else if (region="cz") {b.country="Česká Republika";}
   else {b.country="United Kingdom";}
   b.email=email;
   b.b_date=date;
   b.phone=tel;
   b.favorite=true;
   b.course=specialization;//course();
   b.bg_color=color;
   b.note=comment;
   resarray.push(b);
  
}


export async function onbuttonadd()
{
await task1(10);
task2();
}

export async function piechat()
{
   

// Получение ссылки на элемент canvas в DOM
const $grafica = document.querySelector("#grafica");
// Tags - участки графика
let result1=[
   {country: "Germany"},
   {country: "Germany"},
   {country: "Poland"},
   {country: "Germany"},
   {country: "Ukraine"},
   {country: "Britain"},
]
let ar=result;


if(ar!=null && ar!=undefined && ar.length>0)
{
  ar=_.sortBy(ar,"country")

   const mytags = []
   const mydata =[]
   const mybackgroundColor=[];
   const myborderColor=[];
   let new_country=ar[0].country;
   let counts=1;
   for(let i = 1; i < ar.length ; i ++)
{
     let res=ar[i];
     if (res.country==new_country)
     {
       counts++;
     }
     else
     {
      mytags.push(new_country);
      mydata.push(counts);
      new_country=res.country;
      counts=1;
     }
   }


   mytags.push(new_country);
   mydata.push(counts);



   let val=255 / (mytags.length);
   for(let i = 0; i < mytags.length ; i ++)
   {
      let r=_.random(0, mytags.length);
      let g=_.random(0, mytags.length);
      let b=_.random(0, mytags.length);
      let a1='rgba('+val*r+','+val*g+','+val*b+',0.2)'
      mybackgroundColor.push(a1);
      let a2='rgba('+val*r+','+val*g+','+val*b+',1)'
      
      myborderColor.push(a2);
   }
   const mydataTraffic={data:mydata,backgroundColor:mybackgroundColor,borderColor:myborderColor,borderWidth:1};

   new Chart($grafica, {
      type: 'pie',// Тип графики. Может быть dougnhut  или pie
      data: {
          labels: mytags,
          datasets: [
              mydataTraffic,
              // Больше данных
          ]
      },
  });
}
else{
   const tags = ["Нема даних"]
   const dataTraffic = {
    data: [0], // Данные представляют собой массив, который должен иметь такое же количество значений, как и количество тегов.
    // Теперь цветов фона должно быть столько, сколько данных, т.е. для данного примера 4.
    backgroundColor: [
        'rgba(163,221,203,0.2)'
    ],// Цвет фона
    borderColor: [
        'rgba(163,221,203,1)'
    ],// Цвет границы
    borderWidth: 1,// Толщина границ
   }


new Chart($grafica, {
    type: 'pie',// Тип графики. Может быть dougnhut  или pie
    data: {
        labels: tags,
        datasets: [
            dataTraffic,
            // Больше данных
        ]
    },
});
};

}

export function dayforbirthday(dn)
{
   var now = dayjs();

   const date1 = dayjs(dn);
   let d1=now.diff(date1, 'year') 
   let date2=date1.add(d1, 'year');
   if (now.diff(date1, 'day')>0 ) {date2=date1.add(d1+1, 'year');}

   return Math.floor(date2.diff(now, 'day')); 
}
